src/reactionsystem.jl

```diff
@@ -84,22 +84,25 @@ function mtk_reactions(model::SBML.Model)
             handle_empty_compartment_size = _ -> 1.0)
         symbolic_math = Num(convert(Num, extensive_math,
             convert_time = (x::SBML.MathTime) -> Catalyst.DEFAULT_IV))
+        
+        rstoich = reaction.reactants
+        pstoich = reaction.products
         if reaction.reversible
             symbolic_math = getunidirectionalcomponents(symbolic_math)
             kl_fw, kl_rv = [substitute(x, subsdict) for x in symbolic_math]
-            reagents = getreagents(reaction.stoichiometry, model)
+            reagents = getreagents(rstoich, pstoich, model)
             isnothing(reagents[1]) && isnothing(reagents[2]) && continue
             kl_fw, our = use_rate(kl_fw, reagents[1], reagents[3])
             kl_rv = from_noncombinatoric(kl_rv, reagents[3], our)
             push!(rxs, ModelingToolkit.Reaction(kl_fw, reagents...; only_use_rate=our))
             
-            reagents = getreagents(reaction.stoichiometry, model; rev=true)
+            reagents = getreagents(rstoich, pstoich, model)
```

Comment:
@anandijain: you have to do `reagents = getreagents(pstoich, rstoich, model)` here. This is because we are in the `if reaction.reversible` block. Line 93 does the forward reaction, this line the reverse reaction.